app/code/Magento/Catalog/Controller/Adminhtml/Category/Edit.php

```diff
@@ -114,6 +124,23 @@ public function execute()
     }
 
     /**
+     * Ensure the "scope_data" extension attribute gets added to category
+     *
+     * @return $this
+     */
+    protected function loadCategoryAllScopeData()
+    {
+        $extensionPoolAction = [
+            'read' => [
+                'scopeDataReader' => 'Magento\Catalog\Model\ResourceModel\ReadHandler'
+            ]
+        ];
+        // TODO: Remove hard-coded interface name
+        $this->extensionPool->addAction('Magento\Catalog\Api\Data\CategoryInterface', $extensionPoolAction);
```

Comment:
We initially had the XML code defined in `app/code/Magento/Catalog/etc/adminhtml/di.xml`, but we only wanted the `ReadHandler` class to run when categories were loaded within the context of this controller, so we added the ability to add ExtensionPool items on-the-fly.


This is the code that the above code is replacing:
```xml
    <type name="Magento\Framework\EntityManager\Operation\ExtensionPool">
        <arguments>
            <argument name="extensionActions" xsi:type="array">
                <item name="Magento\Catalog\Api\Data\CategoryInterface" xsi:type="array">
                    <item name="read" xsi:type="array">
                        <item name="scopeDataReader" xsi:type="string">Magento\Catalog\Model\ResourceModel\ReadHandler</item>
                    </item>
                </item>
            </argument>
        </arguments>
    </type>
```